### DISCUSSION TITLE
directus - "DB_CLIENT" Environment Variable is missing.

### DISCUSSION THREAD

**Discussion by karhide-xx** [directus/directus#5575] [Archive]
When I try to test with Directus installed using 'npx create-directus-project test' and then connect this to an empty MariaDB database when I run 'npx directus start' I get the error message:

"DB_CLIENT" Environment Variable is missing.

The operating system is Windows 10 and NPX version 7.12.0 and Node version 14.15.0.

**Comment by rijkvanzanten** [comment]
Could you confirm the `.env` file was created? Did you move any folders after the installation was done? Did you `cd` into the project folder before running start?

**Comment by karhide-xx** [comment]
Yes the .env file was created and contained the details and none on the files were removed.  

Here is the sample from .env file with the user and password removed: 

####################################################################################################
## Database

DB_CLIENT="mysql"
DB_HOST="127.0.0.1"
DB_PORT="3306"
DB_DATABASE="directus2"
DB_USER=[REMOVED]
DB_PASSWORD=[REMOVED]

**Reply by rijkvanzanten** [reply]
Did you cd into the project folder before running start? It sounds like you might've ran `npx directus start` in the wrong folder?

**Comment by karhide-xx** [comment]
Thanks that was the problem.

**Comment by chrisbartley** [comment]
Just a tip for folks (like me) who use a JSON config file for environment variables instead of `.env`.  I got this error when doing the database migration step during a Directus version upgrade because I forgot to explicitly specify the path to the config file.  Fixed it by invoking it like this:

```shell
CONFIG_PATH='./config-dev.js' npx directus database migrate:latest
```
